api/src/main/java/jakarta/servlet/ServletRequest.java

```diff
@@ -359,6 +359,7 @@
      * 
      * @deprecated As of Version 2.1 of the Java Servlet API, use {@link ServletContext#getRealPath} instead.
      */
+    @Deprecated
```

Comment:
This will need a change in the TCK signature tests.

Reply:
Does it mean that at successor of [javax.servlet.sig_4.0_se8:597](https://github.com/eclipse-ee4j/jakartaee-tck/blob/e2c07be584be7a2b9a5f59cfd9651efd2307d944/src/com/sun/ts/tests/signaturetest/signature-repository/javax.servlet.sig_4.0_se8#L597) something like [javax.servlet.sig_4.0_se8:666](https://github.com/eclipse-ee4j/jakartaee-tck/blob/e2c07be584be7a2b9a5f59cfd9651efd2307d944/src/com/sun/ts/tests/signaturetest/signature-repository/javax.servlet.sig_4.0_se8#L666) is done for https://github.com/eclipse-ee4j/servlet-api/blob/1b41595710232879a0a6da86cc21fcba3693723d/api/src/main/java/jakarta/servlet/ServletRequestWrapper.java#L288-L289 once signatures for jakarta.servlet 5.0 are created?

Reply:
We can pick this up when we update the TCK for the javax -> jakarta name change.
This [PR to do the same for EL](https://github.com/eclipse-ee4j/jakartaee-tck/pull/150) should give an idea of what is involved if someone wants to volunteer to do this for Servlet.
We should fix any other binary compatible niggles (e.g. missing generics if any) first.